File: backend/app/schemas/analytics.py
from typing import Literal

from pydantic import BaseModel, Field

from app.schemas.event import EventCause, EventPriority


class CauseBreakdownItem(BaseModel):
    cause: EventCause
    count: int = Field(ge=0)


class PriorityBreakdownItem(BaseModel):
    priority: EventPriority
    count: int = Field(ge=0)


class HourlyTrendItem(BaseModel):
    hour: str
    events: int = Field(ge=0)
    critical: int = Field(ge=0)


class WeeklyTrendItem(BaseModel):
    day: str
    resolved: int = Field(ge=0)
    active: int = Field(ge=0)


class ZoneIntelligenceItem(BaseModel):
    zone: str
    events: int = Field(ge=0)
    risk: int = Field(ge=0, le=100)


class EventAnalyticsData(BaseModel):
    causeBreakdown: list[CauseBreakdownItem]
    priorityBreakdown: list[PriorityBreakdownItem]
    hourlyTrend: list[HourlyTrendItem]
    weeklyTrend: list[WeeklyTrendItem]
    zoneIntelligence: list[ZoneIntelligenceItem]

File: frontend/src/types/analytics.ts
import type { EventCause, EventPriority } from "./event";

export interface CauseBreakdownItem {
  cause: EventCause;
  count: number;
}

export interface PriorityBreakdownItem {
  priority: EventPriority;
  count: number;
}

export interface HourlyTrendItem {
  hour: string;
  events: number;
  critical: number;
}

export interface WeeklyTrendItem {
  day: string;
  resolved: number;
  active: number;
}

export interface ZoneIntelligenceItem {
  zone: string;
  events: number;
  risk: number;
}

/** Response from GET /api/analytics */
export interface AnalyticsResponse {
  causeBreakdown: CauseBreakdownItem[];
  priorityBreakdown: PriorityBreakdownItem[];
  hourlyTrend: HourlyTrendItem[];
  weeklyTrend: WeeklyTrendItem[];
  zoneIntelligence: ZoneIntelligenceItem[];
}

/** @deprecated Use AnalyticsResponse */
export type EventAnalyticsData = AnalyticsResponse;
